libs/common-traits/src/lib.rs

```diff
@@ -222,3 +222,24 @@ pub trait TokenMetadata {
 pub trait TrancheToken<PoolId, TrancheId, CurrencyId> {
 	fn tranche_token(pool: PoolId, tranche: TrancheId) -> CurrencyId;
 }
+
+/// Trait for checking which currencies are "whitelisted" as pool-currencies.
+/// Runtimes need to define which one they want to allow to be
+/// a base currency for a pool.
+pub trait PoolCurrency {
+	type CurrencyId;
+
+	/// Checks whether the given currency is allowed as a
+	/// base currency of a pool
+	fn base(currency: Self::CurrencyId) -> bool;
```

Comment:
Maybe it's clearer to call this method any of `is_valid` / `is_supported` / `is_allowed`?

Reply:
moved to `allowed`. Does this work for you?

Reply:
Yep!